rules/prefer-replace-all.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const getDocumentationUrl = require('./utils/get-documentation-url');
+const quoteString = require('./utils/quote-string');
+
+function isReplaceMethod(node) {
+	return node.callee.property.name === 'replace';
+}
+
+function hasGlobalFlag(node) {
+	const searchPattern = node.arguments[0];
+	return searchPattern && searchPattern.regex && searchPattern.regex.flags === 'g';
+}
+
+function isLiteralCharactersOnly(node) {
+	const searchPattern = node.arguments[0].regex.pattern;
+	const specialCharacters = searchPattern.match(/[$()*+.?[\\\]^{}]/g);
+	const specialCharactersCount = specialCharacters ? specialCharacters.length : 0;
+	const escapedSpecialCharacters = searchPattern.match(/\\[$()*+.?[\\\]^{}]/g);
+	const escapedSpecialCharactersCount = escapedSpecialCharacters ? escapedSpecialCharacters.length : 0;
+	return specialCharactersCount === 2 * escapedSpecialCharactersCount;
+}
+
+function replaceNode(node, fixer) {
+	const stringName = node.callee.object.name;
+	const searchPattern = node.arguments[0].regex.pattern;
+	const replacePattern = node.arguments[1].value;
+	return fixer.replaceText(node, `${stringName}.replaceAll(${quoteString(searchPattern)}, ${quoteString(replacePattern)})`);
+}
+
+function checkNode(context, node) {
+	if (isReplaceMethod(node) && hasGlobalFlag(node) && isLiteralCharactersOnly(node)) {
```

Comment:
Can you also add `arguments` length check here?

Reply:
Also type check for `arguments[1]`

Reply:
Not sure we should check `arguments[1]` has `$0` `$1`

Reply:
Yep, I can add that.

And what you said below is correct, `arguments[1]` shouldn't be changed at all so there's no need to type check. If it contained `$0` or `$1` the regex would have a match group so it would not pass the check for the search phrase containing only literals.